providers/supermicro/firmware_bios_test.go

```diff
@@ -93,7 +94,7 @@ func Test_setComponentUpdateMisc(t *testing.T) {
 					return
 				}
 
-				assert.Nil(t, err)
+				assert.NoError(t, err)
```

Comment:
The use of `require.NoError` and `assert.NoError` seems arbitrary. When to use which is not clear. Is this something that `golangci-lint` catches? If not, I recommend either creating a doc for when one should be used over the other, or just stick to one.